### PR TITLE
test: Code coverage dateRangeModal component

### PR DESCRIPTION
**Pull request by CODEX108**
clickUp: https://app.clickup.com/t/85zrum1n2

image

### CODE REVIEW

File: src/app/fyle/personal-cards/date-range-modal/date-range-modal.component.spec.ts [hunk 1]
```diff
@@ -1,24 +1,83 @@
...
+  it('should dismiss modal when selecting a pre-defined range', () => {
+    const button = getElementBySelector(fixture, '.date-range-modal--label') as HTMLElement;
+    click(button);
+    fixture.detectChanges();
+    expect(modalController.dismiss).toHaveBeenCalledTimes(1);
+  });
+
+  it('should dismiss the modal with custom range dates', () => {
+    const startDate = (component.dateRangeStart.nativeElement.value = '2022-01-01');
+    const endDate = (component.dateRangeEnd.nativeElement.value = '2022-01-15');
```
Comment (Julias0) [inline src/app/fyle/personal-cards/date-range-modal/date-range-modal.component.spec.ts:46]: Please dont do assignment to two different variables in the same statement

**Review by Julias0 — CHANGES_REQUESTED** [review]
minor comment - fix and merge

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/personal-cards/date-range-modal/date-range-modal.component.spec.ts [hunk 2]
```diff
@@ -1,24 +1,83 @@
...
 
-xdescribe('DateRangeModalComponent', () => {
+describe('DateRangeModalComponent', () => {
   let component: DateRangeModalComponent;
   let fixture: ComponentFixture<DateRangeModalComponent>;
+  let modalController: jasmine.SpyObj<ModalController>;
 
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
+  beforeEach(waitForAsync(() => {
```
Comment (Julias0) [inline src/app/fyle/personal-cards/date-range-modal/date-range-modal.component.spec.ts:14]: why is this added to beforeEach?

**Review by Julias0 — APPROVED** [review]

File: src/app/fyle/personal-cards/date-range-modal/date-range-modal.component.spec.ts (hunk 2 above)
Comment (CODEX108) [reply src/app/fyle/personal-cards/date-range-modal/date-range-modal.component.spec.ts:14]: The `async()` method is deprecated and `waitForAsync()` is used instead, therefore the waitForAsync is used in beforeEach.

**Pull request merged**